debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/xstream/XstreamStreamingChangeEventSource.java

```diff
@@ -99,35 +101,18 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
     @Override
     public void commitOffset(Map<String, ?> offset) {
         if (xsOut != null) {
-            try {
                 LOGGER.debug("Recording offsets to Oracle");
                 final LcrPosition lcrPosition = LcrPosition.valueOf((String) offset.get(SourceInfo.LCR_POSITION_KEY));
                 final Long scn = (Long) offset.get(SourceInfo.SCN_KEY);
-                if (lcrPosition != null) {
-                    if (LOGGER.isDebugEnabled()) {
-                        LOGGER.debug("Recording position {}", lcrPosition);
+                final PositionAndScn position = new PositionAndScn(lcrPosition, scn);
+                for (int i = 0; (i < 10) && !lcrMessage.compareAndSet(null, position); i++) {
+                    try {
+                        Thread.sleep(100);
```

Comment:
This is pretty concerning.

Reply:
That's intended - of course would be written with our utils. The key idea is that we can affored to lose few flushes. As soon as the message queue is unblocked a flush will be executed. In this case it might be even unnecessary to do that. As I think about that we can overite it always. As the LCR handler is ublkock it will flush the highest position.